x/concentrated-liquidity/fees.go

```diff
@@ -189,6 +189,17 @@ func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
 		return sdk.Coins{}, err
 	}
 
+	// There may be multiple positions with the same lower and upper tick, but different frozenUntil times.
+	// We retrieve all of them and claim fees for each one.
+	positions, err := k.getAllPositionsWithVaryingFreezeTimes(ctx, poolId, owner, lowerTick, upperTick)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	if len(positions) == 0 {
+		return sdk.Coins{}, cltypes.PositionNotFoundError{PoolId: poolId, LowerTick: lowerTick, UpperTick: upperTick}
+	}
+
```

Comment:
So I reckon that these lines are for checking whether we have a valid position on the owner + lower tick + upper tick combination, do we not check this anyways in the code line below with the feeAccum.hasPosition?

Reply:
Hey, great catch! 

So previously, this was needed because we were iterating over each FullPosition since I made it so that each FullPosition had a fee accumulator. Then, after our call, I realized we only needed a fee accumulator for Position, which then makes this check and the SetPosition requirement in the test not required.

Reply:
Addressed here https://github.com/osmosis-labs/osmosis/pull/4109/commits/838c2ab9855677769da8374c5abbf1dcca93537d